### ISSUE TITLE
Query builder appears not to handle numeric tags properly for InfluxQL queries.

### ISSUE THREAD

**Issue by danatinflux** [grafana/grafana#42980]
When connected to an Influx 2.x data source, the InfluxQL query builder does not properly handle tags with numeric values, as they should have single quotes around them. This appears to tie back to this bit of code:

https://github.com/grafana/grafana/blob/b231afd50fdbb8335cecd61e1e9697b758472c87/public/app/plugins/datasource/influxdb/query_builder.ts#L21

Replication steps:

Create data source:

![image](https://user-images.githubusercontent.com/44617130/145588825-5f1f2944-4423-4ff5-b3b4-a898b4276805.png)

Ingest test data into Cloud 2:

```
gtest3,CODE=404,dc=Seattle fieldval=1 1639146701
gtest3,CODE=200,dc=Tacoma fieldval=1 1639146703
gtest3,CODE=404,dc=Seattle fieldval=1 1639146704
gtest3,CODE=200,dc=Tacoma fieldval=1 1639146706
gtest3,CODE=404,dc=Seattle fieldval=1 1639146708
gtest3,CODE=200,dc=Tacoma fieldval=1 1639146709
gtest3,CODE=404,dc=Seattle fieldval=1 1639146711
gtest3,CODE=200,dc=Tacoma fieldval=1 1639146712
gtest3,CODE=404,dc=Seattle fieldval=1 1639146714
gtest3,CODE=200,dc=Tacoma fieldval=1 1639146716
```
Go to the query builder, and add the 'CODE' tag as the first tag:

![image](https://user-images.githubusercontent.com/44617130/145589355-4f9b812e-d769-430a-ac73-c81891e7f2cb.png)

After that, no other tags can be added. 

However, if other string-value tags are added first, the all-numeric 'CODE' tag can be added at the end:

![image](https://user-images.githubusercontent.com/44617130/145589555-cd40dccb-668b-4e76-a341-b83370fa881a.png)

**Environment**:
- Grafana version: grafana-enterprise-8.3.0~beta2-1.x86_64
- Data source type & version: Influx Cloud 2.0 with InfluxQL query type
- OS Grafana is installed on: CentOS
- User OS & Browser: Firefox
- Grafana plugins: none added

**Comment by gtk-grafana** [comment]
I'm not able to reproduce this on current main branch with the provided use case. @danatinflux are you still having this problem?
<img width="1244" alt="image" src="https://user-images.githubusercontent.com/109082771/185445228-80bbf211-11d7-4598-9eca-a005a0a7c7f3.png">

**Comment by gtk-grafana** [comment]
I am still seeing this on current main
<img width="1254" alt="image" src="https://user-images.githubusercontent.com/109082771/185445597-87fd3901-4882-4f96-9d07-5f4d9c6d10f9.png">

**Comment by gtk-grafana** [comment]
Potential duplicate of [#42980](https://github.com/grafana/grafana/issues/42980), and [4163](https://github.com/grafana/grafana/issues/4163). All issues appear to stem from not knowing if values are string numbers, or number numbers.

Potentially related (thanks to Gabor for the context):
https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/influxdb/influx_query_model.ts#L146

**Comment by gtk-grafana** [comment]
Also related https://github.com/grafana/grafana/issues/40181

**Comment by gtk-grafana** [comment]
So it looks as though influx has changed how they quote values a few times, so changing the current implementation would probably break for a lot of people. How about a setting in the datasource that asks if you want to quote numeric values?

**Comment by gtk-grafana** [comment]
@gabor Do you think that allowing users to set whether their influxdb supports quoted numeric values, and then using the result of that setting to flag [the code](https://github.com/grafana/grafana/commit/b6de656cf1d783b1a3de83b3da70247d0e898020) would work without breaking users' existing dashboards?

Or would it be better to try to add version detection like we're doing for prometheus, and then (if that's possible) specify which versions quote numeric values and which don't?

CC: @evandandrea @itsmylife

**Comment by btasker** [comment]
@gtk-grafana - not sure that the meaning of quotes has changed in InfluxQL (at least not in a meaningful time frame).

Single quote (`'`) delimits a string ([in v0.9](https://archive.docs.influxdata.com/influxdb/v0.9/query_language/spec/#strings) and ) and [in 1.8](https://docs.influxdata.com/influxdb/v1.8/query_language/spec/#strings)), whilst double quote (`"`) indicates a reference to a column/identifier ([in v0.9](https://archive.docs.influxdata.com/influxdb/v0.9/query_language/spec/#identifiers) and [1.8](https://docs.influxdata.com/influxdb/v1.8/query_language/spec/#identifiers)) - the 2.x docs link back to the 1.x spec (see [here](https://docs.influxdata.com/influxdb/v2.0/query-data/influxql/#influxql-reference-documentation) for example).

The only type that a *tag* value can have is string: all tag values can be quoted without impacting the type.

So, you should be fairly safe to quote anything the builder adds as part of a `WHERE` clause, and shouldn't (AFAIK) need to do version detection.

The following (unquoted tag value) won't return results on InfluxDB 2.x or 1.8.x
```
SELECT mean("response_time") AS "mean_response_time" FROM "telegraf"."autogen"."http_response" WHERE time > now() - 2d AND "status_code"=200 GROUP BY time(6h) FILL(null)
```

Nor will it return results with double quotes, only single quotes will work
```
SELECT mean("response_time") AS "mean_response_time" FROM "telegraf"."autogen"."http_response" WHERE time > now() - 2d AND "status_code"='200' GROUP BY time(6h) FILL(null)
```

The behaviour is different for *field* values, because fields can be one of a [number of types](https://docs.influxdata.com/influxdb/v1.8/write_protocols/line_protocol_reference/#data-types)

**Comment by gtk-grafana** [comment]
@btasker Thanks for the context. We can assume that numeric tags can always be quoted, at least for modern versions of influx, however it looks like in versions before 0.1, values were required to be quoted, which was changed in 0.1 where they needed to be unquoted, and then 0.13 they required quotes again, at least according to the [original thread](https://github.com/grafana/grafana/issues/4163) on this. 

I'll open a PR for this and get the ball rolling on testing, my assumption is that the versions potentially impacted are so old they aren't in use anymore, but I want a bit more confidence in that.

Thanks again for the help on this!

**Comment by gtk-grafana** [comment]
A fix for this has been merged and will be released with Grafana 9.2.0! Thank you all for your patience and hard work on this!
